internal/features/features.go

```diff
@@ -30,25 +30,12 @@ const (
 	// the last revision is still the same at the target repository,
 	// and if that is so, skips the reconciliation.
 	OptimizedGitClones = "OptimizedGitClones"
-
-	// GitManagedTransport implements a managed transport for GitRepository
-	// objects that use the libgit2 implementation.
-	//
-	// When enabled, improves the reliability of libgit2 reconciliations,
-	// by enforcing timeouts and ensuring libgit2 cannot hijack the process
-	// and hang it indefinitely.
-	GitManagedTransport = "GitManagedTransport"
```

Comment:
In terms of phasing out feature gates we probably have two options:
1. Completely remove the option (as per the current change).
2. Warn an option is no longer supported and remove it a few minors down the line.

The former is the easiest, but will cause a breaking change as folks upgrading with said options will fail indefinitely. The latter is more work, and will definitely be needed for post GA.

Given how short-lived this feature gate was, and that we are pre-GA. I would favour the former approach, with the caveat that we should add a note on the release.